compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -808,24 +808,30 @@ public void visit(BLangSimpleVariable varNode) {
         BType lhsType = varNode.symbol.type;
         varNode.type = lhsType;
 
-        // Configurable variable type must be a subtype of anydata&readonly.
+        // Configurable variable type must be a subtype of anydata.
         if (configurable && varNode.typeNode != null) {
-            if (!(types.isAssignable(lhsType, symTable.anydataType) &&
-                    types.isAssignable(lhsType, symTable.readonlyType))) {
-                dlog.error(varNode.typeNode.pos,
-                        DiagnosticErrorCode.CONFIGURABLE_VARIABLE_MUST_BE_ANYDATA_AND_READONLY);
-            } else if (!(types.isAssignable(lhsType, symTable.intType) ||
-                    types.isAssignable(lhsType, symTable.floatType) ||
-                    types.isAssignable(lhsType, symTable.stringType) ||
-                    types.isAssignable(lhsType, symTable.booleanType) ||
-                    types.isAssignable(lhsType, symTable.decimalType) ||
-                    types.isAssignable(lhsType, symTable.xmlType) ||
-                    types.isAssignable(lhsType, symTable.arrayType) ||
-                    types.isAssignable(lhsType, symTable.mapAnydataType) ||
-                    types.isAssignable(lhsType, symTable.tableType))) {
-                // TODO: remove this check once runtime support all configurable types
-                dlog.error(varNode.typeNode.pos,
-                        DiagnosticErrorCode.CONFIGURABLE_VARIABLE_CURRENTLY_NOT_SUPPORTED, lhsType);
+            lhsType = getReadOnlyFieldType(varNode.typeNode.pos, lhsType);
+            if (lhsType == symTable.semanticError) {
+                dlog.error(varNode.pos, DiagnosticErrorCode.INVALID_CONFIGURABLE_TYPE, lhsType);
```

Comment:
I don't think we need this error. `lhsType` being a semantic error would mean that there was an issue with that type and that there is at least one error logged for that, right? Fixing the original issue (and only that) would make this error go away, so we aren't adding any value for the user with this error.

Reply:
I have used `getReadOnlyFieldType` method which returns `semanticError` if the type is not selectivelyImmutable. So that's why I added this error log. This is the same behavior wherein the other places `getReadOnlyFieldType` method has been used.

Reply:
Removed the error. Thanx